src/containers/PatientDetails/PatientDocument/index.tsx

```diff
@@ -22,17 +27,85 @@ export interface PatientDocumentProps {
     encounterId?: string;
+    onSubmit?: (formData: QuestionnaireResponseFormData) => Promise<any>;
     onSuccess?: (resource: QuestionnaireResponseFormSaveResponse) => void;
-    autosave?: boolean;
+    onCancel?: () => void;
+    onQRFUpdate?: (questionnaireResponse: QuestionnaireResponse) => void;
+    autoSave?: boolean;
     qrDraftServiceType?: QuestionnaireResponseDraftService;
+    alertComponent?: React.ReactNode | (() => React.ReactNode);
 }
 
 export function PatientDocument(props: PatientDocumentProps) {
-    const { autosave, qrDraftServiceType = 'local' } = props;
+    const { autoSave = false, qrDraftServiceType = 'local', onCancel, onSubmit } = props;
+
+    const params = useParams<{ questionnaireId: string; encounterId?: string }>();
+
+    const { draftQuestionnaireResponseRD, draftInfoMessage, onQRFUpdate, deleteDraft, submitDraft } =
+        useQuestionnaireResponseDraft({
+            subject: `${props.patient.resourceType}/${props.patient.id}`,
+            questionnaireId: props.questionnaireId ?? params.questionnaireId!,
+            questionnaireResponseId: props.questionnaireResponse?.id,
+            qrDraftServiceType,
+            autoSave,
+            questionnaireResponse: props.questionnaireResponse,
+        });
+
+    return (
+        <RenderRemoteData remoteData={draftQuestionnaireResponseRD} renderLoading={Spinner}>
+            {(draftQuestionnaireResponse) => (
+                <PatientDocumentContent
+                    {...props}
+                    key="patient-document-content"
+                    questionnaireResponse={draftQuestionnaireResponse}
+                    onSuccess={async (resource: QuestionnaireResponseFormSaveResponse) => {
+                        await submitDraft();
+                        props.onSuccess && props.onSuccess(resource);
+                    }}
+                    onCancel={async () => {
+                        await deleteDraft();
+                        onCancel?.();
+                    }}
+                    onQRFUpdate={onQRFUpdate}
+                    onSubmit={async (formData) => {
+                        await submitDraft();
+                        return await onSubmit?.(formData);
+                    }}
+                    alertComponent={
+                        <AlertMessage
+                            actionComponent={
+                                <Button
+                                    onClick={async () => {
+                                        await deleteDraft();
+                                    }}
+                                >
+                                    {t`Clear draft`}
+                                </Button>
+                            }
+                            message={draftInfoMessage}
+                        />
+                    }
+                />
+            )}
+        </RenderRemoteData>
+    );
+}
+
+function PatientDocumentContent(props: PatientDocumentProps) {
```

Comment:
Split PatientDocumentProps into PatientDocumentProps and PatientDocumentContentProps (maybe through PatientDocumentBaseProps)
Because now you define too many props that is used only in PatientDocumentContentProps (internal compoennt), but not in PatientDocumentProps that we expose as API, e.g. alert component